### PR TITLE
Add Var.read_var for a vector of NetCDF files and update SimDir to handle mulitiple outputs

### LINKED ISSUE #115: Extend ClimaAnalysis to stitch together data in multiple directories
Using the `aggdim` option in NetCDF, we can create individual datasets from multiple files stitched together along the time dimension. This capability will be needed to properly support restarted simulations.

### PR DESCRIPTION
**Pull request by ph-kev**
closes #115 - This PR adds `Var.read_var` for a vector of NetCDF files and update `SimDir` to handle multiple outputs. 

For instance, the file directory could look like this:
```
global_diagnostics/
├── output_0001
│   ├── pfull_2.0d_inst.nc
│   ├── ts_1.0h_max.nc
│   └── ts_10d_average.nc
├── output_0002
│   ├── ts_1.0h_max.nc
│   └── ts_10d_average.nc
└── output_0003
    └── pfull_2.0d_inst.nc
```
The following code would work:
```julia
simulation_path = joinpath(simulation_path)
ts_max_var = get(simdir, short_name = "ts", reduction = "max", period = "1.0h")
pfull_var = get(simdir, short_name = "pfull", reduction = "2.0d", period = "inst")
```

When checking if it is okay to aggregate dimensions, `read_var` only checks for
- if the time dimension exists and the names of the time dimensions are all the same,
- if the start dates are the same,
- if the times are in increasing order if aggregated.

### CODE REVIEW

File: src/Sim.jl [hunk 1]
```diff
@@ -33,36 +33,51 @@ function SimDir(simulation_path::String)
...
+                    # Do not sort because walkdir will give the files in top-down order
+                    if variable_reduction_period[period] isa Vector
+                        push!(variable_reduction_period[period], full_path)
+                    else
+                        variable_reduction_period[period] =
+                            [variable_reduction_period[period], full_path]
+                    end
+                else
+                    push!(variable_reduction_period, period => full_path)
+                end
```
Comment (Sbozzolo) [inline src/Sim.jl:63]: I'd suggest that we always store file paths in a vector of strings, even when we only have one.

File: src/Sim.jl [hunk 2]
```diff
@@ -33,36 +33,51 @@ function SimDir(simulation_path::String)
...
+                # Add to allfiles
+                push!(allfiles, full_path)
+
+                # At the end we have three layers of dictionaries.
+                #
+                # The first layer maps variable short names to a dictionary, which maps
+                # reductions to a dictionary, which maps periods to the path of the file that
+                # contains that chain
+                #
+                # Example: variable_paths = {"ta" : {"max": {"6.0h" => file.nc}}}
```
Comment (Sbozzolo) [inline src/Sim.jl:79]: This comment should be updated for the general case with multiple paths.

Also, what does "chain" mean in this context? (I probably wrote this but I don't understand)

File: src/Var.jl [hunk 3]
```diff
@@ -282,6 +282,105 @@ function read_var(path::String; short_name = nothing)
...
+    # Assume that `paths` contain all the datasets we want to try to stitch together
+    # Assume that `paths` is sorted (will rely on simdir to return it sorted)
+
+    # Helper function to find the name of the time dimension
+    function find_name(dim_names)
+        for dim_name in dim_names
+            dim_name in TIME_NAMES && return dim_name
+        end
+        return nothing
+    end
```
[the remark this answers is not in the document]
Comment (Sbozzolo) [reply src/Var.jl:323]: (Without errors I guess)

[the remark this answers is not in the document]
Comment (ph-kev) [reply src/Var.jl:323]: It is almost `time_name`, but the `time_name` function is implemented for `OutputVar`. I could use `find_dim_name` and use a `try` and `catch` if that is better.

File: src/Sim.jl (hunk 2 above)
Comment (ph-kev) [reply src/Sim.jl:79]: I don't know what "chain" means either. I am going to remove that phrasing.

File: src/Var.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sbozzolo) [reply src/Var.jl:323]: I think what you have is fine

**Review by Sbozzolo — APPROVED** [review]
Thank you! This is very useful!

**Pull request merged**